third_party/tonic/converter/dart_converter.h

```diff
@@ -279,15 +281,80 @@ struct DartConverter<const char*> {
 ////////////////////////////////////////////////////////////////////////////////
 // Collections
 
+inline Dart_Handle LookupNonNullableType(const std::string& library_name,
+                                         const std::string& type_name) {
+  auto library =
+      Dart_LookupLibrary(DartConverter<std::string>::ToDart(library_name));
+  if (LogIfError(library)) {
+    return library;
+  }
+  auto type_string = DartConverter<std::string>::ToDart(type_name);
+  if (LogIfError(type_string)) {
+    return type_string;
+  }
+  auto type = Dart_GetNonNullableType(library, type_string, 0, nullptr);
+  if (LogIfError(type)) {
+    return type;
+  }
+  return type;
+}
+
+template <typename T,
+          std::enable_if_t<std::is_same<std::string, T>::value, int> = 0>
+Dart_Handle ToDartTypeHandle() {
+  return LookupNonNullableType("dart:core", "String");
+}
+
+template <typename T, std::enable_if_t<std::is_integral<T>::value, int> = 0>
+Dart_Handle ToDartTypeHandle() {
+  return LookupNonNullableType("dart:core", "int");
+}
+
+template <typename T,
+          std::enable_if_t<std::is_floating_point<T>::value, int> = 0>
+Dart_Handle ToDartTypeHandle() {
+  return LookupNonNullableType("dart:core", "double");
+}
+
```

Comment:
Not sure how frequently ToDartTypeHandle() is called for these basic types (string, int, double), depending on that it might make sense to cache these types and avoid the need to lookup the library and get the non nullable type everytime.

Reply:
Maybe not for this PR but something to consider for future.